src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.ts

```diff
@@ -50,10 +50,10 @@ export default class PaypalCommercePaymentProcessor {
         const buttonParams: ButtonsOptions = {
             ...params,
             createOrder: () => this._setupPayment(cartId, paramsForProvider),
-            onClick: data => {
+            onClick: async (data, actions) => {
                 this._fundingSource = data.fundingSource;
 
-                params.onClick?.(data);
+                return params.onClick?.(data, actions);
```

Comment:
`onClick` doesn't seem to return a promise (unlike `onValidate`). So there's probably no need for `async` or call `return`.

Reply:
It depends on purpose. You can use it with promise too
Example from PayPal here https://developer.paypal.com/docs/checkout/integration-features/validation/#asynchronous-validation